baseplate/clients/thrift.py

```diff
@@ -166,6 +169,21 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
                     if span.flags:
                         prot.trans.set_header(b"Flags", str(span.flags).encode())
 
+                    min_timeout = None
+                    is_time_retry_policy = (
+                        retry_policy is TimeBudgetRetryPolicy
+                        or retry_policy is ExponentialBackoffRetryPolicy
+                    )
+                    if is_time_retry_policy:
+                        min_timeout = value
+                    if self.pool.timeout:
+                        if not min_timeout or self.pool.timeout < min_timeout:
+                            min_timeout = self.pool.timeout
+                    if min_timeout:
```

Comment:
nit: I don't think it's possible but we probably want to check that `min_timeout` is not negative? In baseplate.go we checked if it's less than 1ms (which includes negative values) and give it a minimal of 1ms (https://github.com/reddit/baseplate.go/blob/08f6b3c8e11a95dcc6a99d26d8b961acd6c08615/thriftbp/client_middlewares.go#L234). the `ceil` part handled the case that the timeout is between 0 and 1ms, but doesn't handle the case if it's negative.